### ISSUE TITLE
The "--optimize-autoloader" option does not exist.

### ISSUE THREAD

**Issue by arnoldbird** [hechoendrupal/drupal-console#3060]
When I run...

`composer require drupal/console:~1.0 --prefer-dist --optimize-autoloader`

...I get...
```
   [RuntimeException]                                  
  The "--optimize-autoloader" option does not exist. 
```

**Comment by jmolivas** [comment]
@arnoldbird  Which `composer` version do you have ?

Try `composer --version` to find out and ` composer self-update` to update to latest version.

**Comment by arnoldbird** [comment]
root@ryandw-desktop:~# composer --version
Composer version @package_branch_alias_version@ (@package_version@) @release_date@
root@ryandw-desktop:~# cd /var/www/d8/public_html/
root@ryandw-desktop:/var/www/d8/public_html# composer self-update
  [InvalidArgumentException]             
  Command "self-update" is not defined.  
root@ryandw-desktop:/var/www/d8/public_html# cd ~
root@ryandw-desktop:~# composer self-update                                  
  [InvalidArgumentException]             
  Command "self-update" is not defined.

**Comment by jmolivas** [comment]
@arnoldbird can you try re-installing composer?

**Comment by arnoldbird** [comment]
Thanks.  Re-installing composer fixed it.

To remove composer, I looked for composer.phar on my system with the find command and could not find it.  So I deleted the /root/.composer directory and re-installed as shown at https://getcomposer.org/download/

I then moved the composer.phar to /usr/local/bin/composer

Logged out of root account and then...
sudo /usr/local/bin/composer require drupal/console:~1.0 --prefer-dist --optimize-autoloader

And the install went smoothly.  Console works now.
